base/src/main/java/org/kontinuity/catapult/base/identity/IdentityVisitor.java

```diff
@@ -0,0 +1,15 @@
+package org.kontinuity.catapult.base.identity;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public interface IdentityVisitor {
+
+    default void visit(TokenIdentity token) {
```

Comment:
Interesting design pattern.  Is there a way that accomplishes this more typesafe that doesn't lead to a contract requiring both username/password and token when only one would be supported at a time?  `UnsupportedOpersationException` is usually used by last resort; unsupported operations should be blocked by the compiler whenever possible.